src/tree.rs

```diff
@@ -695,6 +695,71 @@ impl<T> Tree<T> {
         Result::Ok(())
     }
 
+    /// Moves the node to a position amongst sibling nodes.
+    ///
+    /// Any children will remain attached to this node.
+    ///
+    /// ```
+    /// use id_tree::*;
+    /// use id_tree::InsertBehavior::*;
+
+    /// let mut my_tree = TreeBuilder::<i32>::new()
+    ///         .with_node_capacity(5)
+    ///         .build();
+
+    /// let root_id: NodeId = my_tree.insert(Node::new(0), AsRoot).unwrap();
+    /// let c1: NodeId = my_tree.insert(Node::new(1), UnderNode(&root_id)).unwrap();
+    /// let _c2 = my_tree.insert(Node::new(2), UnderNode(&root_id)).unwrap();
+    /// let _c3 = my_tree.insert(Node::new(3), UnderNode(&root_id)).unwrap();
+    /// let _c4 = my_tree.insert(Node::new(4), UnderNode(&root_id)).unwrap();
+
+    /// for (i,n) in my_tree.children(&root_id).unwrap().enumerate() {
+    ///     println!("i={} n={:?}", i, n.data());
+    /// }
+    
+    /// my_tree.make_nth_sibling(&c1, 3).unwrap();
+
+    /// for (i,n) in my_tree.children(&root_id).unwrap().enumerate() {
+    ///     println!("i={} n={:?}", i, n.data());
+    /// }
+    /// ```
+    ///
+
+    pub fn make_nth_sibling(&mut self, node: &NodeId, pos: usize) -> Result<(), NodeIdError> {
+
+        let parent = self.get(node)?.parent()
+            .ok_or(NodeIdError::NodeIdNoLongerValid)?
+            .clone();
+
+        let num_children = self.children_ids(&parent)?.count();
+        if pos >= num_children {
```

Comment:
I've been going back and forth on what the behavior should be here when `pos >= num_children`.

I think right now I'm leaning towards a slightly different approach where we don't return an error at all in this case. If we created a new variable like this instead:
```
let target_pos = pos.min(num_children - 1);
```
we could then continue along with `target_pos` from that point on knowing that it's a safe index to use when swapping below.

I _think_ I prefer this approach instead of returning a `Err(NodeIdError::NodeIdNoLongerValid)` since that's not _technically_ what the problem is here.

If it's important that we return an `Err(...)` here the I'm afraid we might need to add a new variant to `NodeIdError` so that we can more accurately represent what went wrong. Unfortunately that change would require a major version bump because `NodeIdError` isn't currently marked with `#[non_exhaustive]`. (That doesn't mean we can't do that if we need to).